### PR TITLE
Add credential method to access GitHub with Personal Access Token

### LINKED ISSUE #739: Use github access token for git mirroring
Hello,

The official document describes that there are 2 ways to authenticate to access git repository when git mirroring, which are password-based authentication and OpenSSH key.

However, I wonder if we could use github access token as authentication key
- github access token: https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token

### PR DESCRIPTION
**Pull request by taiga-nada**
Motivation:

It'd be nice if a user can use [GitHub Personal Access Token (PAT)](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) to mirror a GitHub repository.

Modifications:

- Added `AccessTokenMirrorCredential` to access a GitHub repository with PAT.
- Added documentation.

Result:

- Closes #739

### CODE REVIEW

**Review by trustin — COMMENTED** [review]
So far so good! Could you continue to:
- Add documentation; and
- Update [GitMirrorAuthTest](https://github.com/line/centraldogma/blob/e93bc5a04eb3d38a0f315d9c5fdb0db3b0af1f52/it/src/test/java/com/linecorp/centraldogma/it/mirror/git/GitMirrorAuthTest.java#L53) so we can test access-token-based authentication manually?

**Comment by trustin** [comment]
@taiga-nada will send a follow-up PR that updates `GitMirrorAuthTest` that tests access-token-based authentication. This PR will be ready for review after adding documentation.

**Comment by taiga-nada** [comment]
I updated documentations.

**Review by trustin — APPROVED** [review]
🚀🚀🚀 (Build failure will be fixed once you merge the latest master)

File: site/src/sphinx/mirroring.rst
Comment (ikhoon) [inline site/src/sphinx/mirroring.rst:219]: Let's link https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token to `GitHub Personal Access Token`

**Review by ikhoon — APPROVED** [review]
Thanks, @taiga-nada! 🚀💯 LGTM if CI builds pass.

File: site/src/sphinx/mirroring.rst
Comment (taiga-nada) [reply site/src/sphinx/mirroring.rst:219]: OK. I'll do it!

Comment (taiga-nada) [reply site/src/sphinx/mirroring.rst]: @ikhoon I added at [4afa19b](https://github.com/line/centraldogma/pull/813/commits/4afa19be8184425bc3b237b60be495ab267668c7)

**Review by jrhee17 — APPROVED** [review]
Looks good! Thanks 🙇 👍 🚀

**Pull request merged**
